src/bin/mosaic-libraries.ts

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env node
+
+import * as commander from 'commander';
+
+import { version } from '../../package.json';
+import Logger from '../Logger';
+import OriginChain from "../NewChain/OriginChain";
+import Web3 = require("web3");
+import MosaicConfig from "../Config/MosaicConfigV2";
+
+let mosaic = commander
+  .version(version)
+  .arguments('<chain> <origin-websocket> <deployer>');
+mosaic.action(
+  async (
+    chain: string,
+    originWebsocket: string,
+    deployer: string,
+  ) => {
+
+    try {
+
+      const originWeb3 = new Web3(originWebsocket);
+      const {
+        gatewayLib,
+        messageBus,
+        merklePatriciaProof,
+      } = await OriginChain.deployLibraries(originWeb3, deployer);
+
+      const mosaicConfig:MosaicConfig = new MosaicConfig();
+
+      mosaicConfig.originChain.chain = chain;
+      mosaicConfig.originChain.contractAddress.gatewayLibAddress = gatewayLib.address;
+      mosaicConfig.originChain.contractAddress.messageBusAddress = messageBus.address;
+      mosaicConfig.originChain.contractAddress.merklePatricialLibAddress = merklePatriciaProof.address;
+
+      mosaicConfig.writeToMosaicConfigDirectory();
```

Comment:
It seems you are overwriting any existing mosaic config. Instead, you should append to any existing config.

Reply:
Done